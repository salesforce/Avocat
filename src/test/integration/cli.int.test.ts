import {TestCliUtils} from '../utils/test-cli-utils';

describe('CLI Test', () => {
    describe('Help Command Test', () => {
        describe('When running avocat with help option', () => {
            it('Shows available options and commands', () => {
                jest.setTimeout(30000);
                return TestCliUtils.transpileAndRunAvocatFromCLI(['--help']).then(result => {
                    expect(result.code).toBe(0);
                    expect(result.stdout.trim()).toMatchSnapshot();
                }).catch(console.error);
            });
        });
    });
});

