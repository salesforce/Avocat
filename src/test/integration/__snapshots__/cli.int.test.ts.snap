// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CLI Test Help Command Test When running avocat with help option Shows available options and commands 1`] = `
"Usage: avocat [command] [options]

ðŸ¥‘ Continuous contract testing for HTTP APIs

Options:
  -v, --version           Avocat current version
  -h, --help              Show available commands/options

Commands:
  status
  import <contract_path>"
`;
